modules/appeals_api/app/models/appeals_api/supplemental_claim.rb

```diff
@@ -266,12 +253,47 @@ def decision_date_not_in_past(issue, issue_index)
       add_decision_date_error "isn't in the past: #{issue.decision_date_string.inspect}", issue_index
     end
 
+    # rubocop:disable Layout/LineLength
     def add_decision_date_error(string, issue_index)
-      add_error "included[#{issue_index}].attributes.decisionDate #{string}"
+      add_error("included[#{issue_index}].attributes.decisionDate #{string}", title: 'Invalid field value',
+                                                                              missing_fields: ['decision date'],
+                                                                              pointer: '/data/attributes/included/items/attributes/properties',
+                                                                              code: 144)
+    end
+    # rubocop:enable Layout/LineLength
+
+    # validation (header & body)
+    # Schemas take care of most of the requirements, but we need to check that both header & body data is provided
+    def required_claimant_data_is_present
+      has_claimant_headers = claimant.first_name.present?
+      # form data that includes a claimant is also sufficient to know it's passed the schema
+      has_claimant_data = data_attributes&.fetch('claimant', nil).present?
+
+      return if !has_claimant_headers && !has_claimant_data # No claimant headers or data? not a problem!
+      return if has_claimant_headers && has_claimant_data # Has both claimant headers and data? A-ok!
+
+      unless has_claimant_headers
+        add_error('Claimant data was provided but missing claimant headers', missing_fields: missing_fields,
+                                                                             title: 'Missing required fields',
+                                                                             pointer: '/',
+                                                                             code: 145)
+      end
+
+      unless has_claimant_data
+        add_error('Claimant headers were provided but missing claimant data', missing_fields: missing_fields,
+                                                                              title: 'Missing required fields',
+                                                                              pointer: '/data/attributes',
+                                                                              code: 145)
+      end
     end
 
-    def add_error(message)
-      errors.add(:base, message)
+    def address_combined
+      return unless veteran&.dig('address', 'addressLine1')
```

Comment:
I'm surprised this worked - I thought the `veteran` method returned an Appellant that doesn't implement `dig`. 🤔 What am I missing here?

Reply:
@cilestin actually, that was already present. I deleted the method 'add_error', and it moved the method that you are referring to, up a few lines.